File: antson.csharp/antson.csharp/semantics/io.odysz/x/TransException.cs
namespace io.odysz.transact.x
{
	[System.Serializable]
	public class TransException : System.Exception
	{
		public TransException(string format, params object[] args)
			: base(string.IsNullOrEmpty(format) ?  null
				  : args != null && args.Length > 0 ?
					string.Format(format, args) : format)
		{
		}
	}
}

File: antson.java/src/test/java/io/odysz/semantic/jprotocol/test/TransException.java
package io.odysz.semantic.jprotocol.test;

import io.odysz.common.LangExt;

public class TransException extends Exception {
	private static final long serialVersionUID = 1L;

	public TransException(String format, Object... args) {
		super(LangExt.isblank(format) ? null
			: args != null && args.length > 0 ?
					String.format(format, args) : format);
	}
}
